src/Casbin.AspNetCore/ServiceCollectionExtension.cs

```diff
@@ -1,13 +1,49 @@
-﻿using Microsoft.Extensions.DependencyInjection;
+﻿using System;
+using Casbin.AspNetCore.Abstractions;
+using Casbin.AspNetCore.Policy;
+using Casbin.AspNetCore.Transformers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using NetCasbin;
+using NetCasbin.Model;
 
 namespace Casbin.AspNetCore
 {
     public static class ServiceCollectionExtension
     {
-        public static IServiceCollection AddCasbinAuthorization(this IServiceCollection services)
+        public static IServiceCollection AddCasbinAuthorizationCore(this IServiceCollection services, Action<CasbinAuthorizationCoreOptions> configure)
         {
-
+            var options = CasbinAuthorizationCoreOptions.GetDefault();
+            configure(options);
+            CheckOptions(options);
+            services.TryAdd(ServiceDescriptor.Describe(typeof(Model),
+                p => options.ModelFactory?.Invoke(),
+                options.ModelLifeTime));
+            services.TryAdd(ServiceDescriptor.Describe(typeof(Enforcer),
```

Comment:
`new Enforcer` will only be executed once right?

Reply:
If you use the scoped lifetime, it will be executed once at the beginning of the scope. If you create it without model it will load all the policies here.

Reply:
I found you are using a scoped lifetime, so I'm afraid every time a request comes in, will execute `new Enforcer` which I think may not necessary. A singleton may be better?

Reply:
The model will have two lifetimes. and when use singleton lifetime we will not edit safely the policy because the policy is not a thread safety container.
Some user need the scoped option, like [this issue ](https://github.com/casbin/Casbin.NET/issues/15).

Reply:
@Sagilio  @Joey If there is a mechanism that can subscribe the policy modification and update current loaded policies, I rather use 
 Sigleton lifetime service than reloading it every request/invoke for better performance.

Reply:
@arthuridea The IWatcher at Casbin.NET can do it, but you will also need concurrent container support when you use it at asp net core.
I will enhance them later and change the default option to Singleton when it can be used safely.

Reply:
@Sagilio when I first use Casbin.NET ,there's no netCore version of IWatcher yet.I simplely use singleton cache dropped when policy modified.but I didn't take much time on data consistancy and concurrency ensurence as the app requires less reliable on that.hopes a robust library can be created